File: 卡码/代码随想录/C++/二叉树/02-二叉树属性/LeetCode_0104.cpp
//104. 二叉树的最大深度
#include <iostream>
#include "../TreeNode.h"
using namespace std;

class Solution {
public:
  int maxDepth(TreeNode* root) {
  	if(root==nullptr) return 0;
  	return 1+max(maxDepth(root->left),maxDepth(root->right));
  }
};

int main(){
  system("pause");
  return 0;
}

File: 卡码/代码随想录/1/06-二叉树/Java/binaryTree/src/main/java/com/kurumi/binarytree/_02_二叉树属性/LeetCode_0104.java
//104. 二叉树的最大深度
package com.kurumi.binarytree._02_二叉树属性;

import com.kurumi.binarytree.TreeNode;

public class LeetCode_0104 {

}
class Solution {
  public int maxDepth(TreeNode root) {
    if(root==null) return 0;
    return 1+Integer.max(maxDepth(root.left),maxDepth(root.right));
  }
}
